src/ext/java/org/opentripplanner/ext/siri/AddedTripBuilder.java

```diff
@@ -0,0 +1,379 @@
+        isJourneyPredictionInaccurate,
+        calls.get(stopSequence),
+        occupancy
+      );
+    }
+
+    if (cancellation || stopPattern.isAllStopsCancelled()) {
+      updatedTripTimes.cancelTrip();
+    } else {
+      updatedTripTimes.setRealTimeState(RealTimeState.ADDED);
+    }
+
+    /* Validate */
+    var validityResult = updatedTripTimes.validateNonIncreasingTimes();
+    if (validityResult.isFailure()) {
+      return validityResult.toFailureResult();
+    }
+
+    // Adding trip to index necessary to include values in graphql-queries
+    // TODO - SIRI: should more data be added to index?
+    transitModel.getTransitModelIndex().getTripForId().put(tripId, trip);
+    transitModel.getTransitModelIndex().getPatternForTrip().put(trip, pattern);
+    transitModel.getTransitModelIndex().getPatternsForRoute().put(route, pattern);
+
+    return Result.success(new TripUpdate(stopPattern, updatedTripTimes, serviceDate));
+  }
+
+  /**
+   * Method to create a Route. Commonly used to create a route if a realtime message
+   * refers to a route that is not in the transit model.
+   *
+   * We will find the first Route with same Operator, and use the same Authority
+   * If no operator found, copy the agency from replaced route
+   *
+   * If no name is given for the route, an empty string will be set as the name.
+   *
+   * @return a new Route
+   */
+  private Route createRoute() {
+    var routeBuilder = Route.of(entityResolver.resolveId(lineRef));
+
+    routeBuilder.withShortName(shortName);
+    routeBuilder.withMode(transitMode);
+    routeBuilder.withNetexSubmode(transitSubMode);
+    routeBuilder.withOperator(operator);
+
+    // TODO - SIRI: Is there a better way to find authority/Agency?
+    Agency agency = transitModel
+      .getTransitModelIndex()
+      .getAllRoutes()
+      .stream()
+      .filter(r -> r != null && r.getOperator() != null && r.getOperator().equals(operator))
+      .findFirst()
+      .map(Route::getAgency)
+      .orElseGet(() -> replacedRoute != null ? replacedRoute.getAgency() : null);
+    routeBuilder.withAgency(agency);
+
+    return routeBuilder.build();
+  }
+
+  private Trip createTrip(Route route, FeedScopedId calServiceId) {
+    var tripBuilder = Trip.of(tripId);
+    tripBuilder.withRoute(route);
+
+    // Explicitly set TransitMode on Trip - in case it differs from Route
+    tripBuilder.withMode(transitMode);
+    tripBuilder.withNetexSubmode(transitSubMode);
+
+    tripBuilder.withServiceId(calServiceId);
+
+    // Use destinationName as default headsign - if provided
+    tripBuilder.withHeadsign(NonLocalizedString.ofNullable(headsign));
+
+    tripBuilder.withOperator(operator);
+
+    // TODO - SIRI: Populate these?
+    tripBuilder.withShapeId(null); // Replacement-trip has different shape
+    //        trip.setTripPrivateCode(null);
+    //        trip.setTripPublicCode(null);
+    tripBuilder.withGtfsBlockId(null);
+    tripBuilder.withShortName(null);
```

Comment:
What do these lines achieve?

Reply:
Nothing, they should be removed. They are carried over from existing code.